rspace/src/test/scala/coop/rchain/rspace/history/HistorySpec.scala

```diff
@@ -100,22 +239,45 @@ class HistorySpec extends FlatSpec with Matchers with OptionValues with InMemory
       case Skip(affix, _) => affix.toSeq.toList shouldBe bytes
       case p              => fail("unknown trie prefix" + p)
     }
+
+  def randomKey(size: Int) = List.fill(size)((Random.nextInt(256) - 128).toByte)
+
+  def generateRandomInsert(size: Int) = List.fill(size)(insert(randomKey(32)))
+  def generateRandomDelete(size: Int) = List.fill(size)(delete(randomKey(32)))
+  def generateRandomDeleteFromInsert(size: Int, inserts: List[InsertAction]) =
+    Random.shuffle(inserts).take(size).map(i => delete(i.key))
+
+  def updateState(state: TrieMap[KeyPath, Blake2b256Hash], actions: List[HistoryAction]) = {
+    actions.map(
+      a =>
+        a match {
+          case InsertAction(key, value) => state.put(key, value)
+          case DeleteAction(key)        => state.remove(key)
+        }
+    )
```

Comment:
Pattern matching can be done directly with anonymous functions without explicitly specifying argument and _match_.
https://www.scala-lang.org/files/archive/spec/2.12/08-pattern-matching.html#pattern-matching-anonymous-functions
```scala
actions.map {
  case InsertAction(key, value) => state.put(key, value)
  case DeleteAction(key)        => state.remove(key)
}
```